algorithms/unit_tests/CMakeLists.txt

```diff
@@ -35,6 +35,7 @@ foreach(Tag Threads;Serial;OpenMP;Cuda;HPX;HIP;SYCL;OpenMPTarget)
       "#include <Test${Tag}_Category.hpp>\n"
       "#include <TestRandomCommon.hpp>\n"
       "#include <TestSortCommon.hpp>\n"
+      "#include <TestNestedSort.hpp>\n"
```

Comment:
Why is it not a separate cpp source file?

Reply:
I would prefer generating multiple source file but this do not need to be handled here if you open an issue for it